### PR TITLE
feat: human-readable issue handles (title-slug + short-id)

### PR DESCRIPTION
**Pull request by harshanandak**
## Human-readable issue handles (kernel 1db53c60)

Opaque UUIDs make it hard to tell what an issue is or track where an agent is across sub/dependency tasks. This adds a readable **handle** — `<title-slug>-<short-id>`, e.g. `add-oauth-login-56a3a16d`.

### What changed
- **`issueHandle(issue)`** in `lib/issue-render.js` — `slugify(first 4 title words) + '-' + 8-char short id`; degrades to the bare short id when there's no usable title.
- **Shown** in `forge ready`/`list` (ID column) and in write confirmations at an interactive terminal (`✓ Created add-oauth-login-56a3a16d`).
- **Resolvable** (`lib/kernel/issue-id-resolver.js`): `forge show add-oauth-login-56a3a16d`, `forge show 56a3a16d`, or even a **stale slug + right id** all resolve — the resolver reads the trailing 8-hex short id; the slug is display sugar.

### Safety
- Canonical id stays the full UUID (stable, kernel-native).
- Short legacy/imported ids (`forge-2a3bc9`, `bd-abc`, <8 hex suffix) **pass through unchanged with no extra lookup** — the handle form requires ≥8 trailing hex (matching `shortId`), and an exact whole-token match is preferred first so imported handle-shaped ids (`legacy-2a3bc9de`) still resolve exactly.

### Verification
- 62 issue/render/resolver tests incl. handle unit + resolution + legacy-passthrough safety; full commands/bin/kernel/e2e sweep green (1262 pass); lint clean; D20 release gate PASS.
- End-to-end: `forge ready` shows handles; `forge show`/`claim <handle>` resolve; wrong-slug-right-id resolves.

Folded into the 0.1.0 beta per maintainer request.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added human-readable issue handles in the format `title-slug-short-id` across issue lists, ready views, and interactive success confirmations.
  * Issue lookup now accepts full UUIDs, short IDs, and matching human-readable handles (for resolving issues in show commands).

* **Bug Fixes**
  * Improved resolution for handle-like tokens, including edge cases for hex-like slugs and ambiguous/too-short suffixes.
  * When title details aren’t available, the display cleanly falls back to the short ID.

* **Tests**
  * Expanded unit/integration coverage for handle formatting and identifier resolution behavior.

### CODE REVIEW

File: lib/kernel/issue-id-resolver.js [hunk 1]
```diff
@@ -94,11 +94,32 @@ function ambiguousError(token, candidates) {
...
+		if (wholeExact) return { id: wholeExact.id };
+		ref = handle[1];
+	}
+
+	const isPrefix = HEX_PREFIX_PATTERN.test(ref);
+	const isShortHex = !isPrefix && SHORT_HEX_PATTERN.test(ref);
 	if (!isPrefix && !isShortHex) return { id: token };
 
-	const needle = token.toLowerCase();
+	const needle = ref.toLowerCase();
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply lib/kernel/issue-id-resolver.js]: ✅ Fixed (eb16a697). Confirmed the bug: a title of all-hex-letter words (e.g. 'Facade decade fee add') yields a handle that matches HEX_PREFIX_PATTERN, so the extraction branch was skipped. Now: extracted resolveHexToken(); when the whole hex-ish token resolves to nothing, retry with its trailing 8-hex short id. Partial dashed UUID prefixes (no 8-hex suffix, e.g. deadbeef-1111) and legacy/imported ids are unaffected. Added regression tests + verified end-to-end (forge show facade-decade-fee-add-<id> now resolves). (The OpenGrep exec note is a false positive — this resolver has no child_process usage; lookup is injected.)
